include/GIL/InstBase.hpp

```diff
@@ -0,0 +1,235 @@
+    Operand(llvm::APFloat literalFloat) : kind(OperandKind::LiteralFloatKind), data(std::move(literalFloat)) {}
+    Operand(llvm::StringRef literalString) : kind(OperandKind::LiteralStringKind), data(std::move(literalString)) {}
+    // Operand(Function *symbol) : kind(OperandKind::SymbolKind), data(symbol) {}
+    // Operand(Type type) : kind(OperandKind::TypeKind), data(std::move(type)) {}
+    // Operand(Member member) : kind(OperandKind::MemberKind), data(std::move(member)) {}
+    Operand(BasicBlock *label) : kind(OperandKind::LabelKind), data(label) {}
+    ~Operand() {
+        switch (kind) {
+        case OperandKind::ValueKind: data.value.~GILValue(); break;
+        case OperandKind::LiteralIntKind: data.literalInt.~APInt(); break;
+        case OperandKind::LiteralFloatKind: data.literalFloat.~APFloat(); break;
+        case OperandKind::LiteralStringKind: data.literalString.~StringRef(); break;
+        case OperandKind::SymbolKind: break;
+        case OperandKind::TypeKind: // data.type.~Type(); break;
+        case OperandKind::MemberKind: // data.member.~Member(); break;
+        case OperandKind::LabelKind: break;
+        }
+    }
+
+    OperandKind getKind() { return kind; }
+
+    GILValue getGILValue() {
+        assert(kind == OperandKind::ValueKind && "Operand is not a value");
+        return data.value;
+    }
+    llvm::APInt getLiteralInt() {
+        assert(kind == OperandKind::LiteralIntKind && "Operand is not a literal integer");
+        return data.literalInt;
+    }
+    llvm::APFloat getLiteralFloat() {
+        assert(kind == OperandKind::LiteralFloatKind && "Operand is not a literal float");
+        return data.literalFloat;
+    }
+    llvm::StringRef getLiteralString() {
+        assert(kind == OperandKind::LiteralStringKind && "Operand is not a literal string");
+        return data.literalString;
+    }
+    // Function *getSymbol() {
+    //     assert(kind == OperandKind::SymbolKind && "Operand is not a symbol");
+    //     return data.symbol;
+    // }
+    // Type getType() {
+    //     assert(kind == OperandKind::TypeKind && "Operand is not a type");
+    //     return data.type;
+    // }
+    // Member getMember() {
+    //     assert(kind == OperandKind::MemberKind && "Operand is not a member");
+    //     return data.member;
+    // }
+    BasicBlock *getLabel() {
+        assert(kind == OperandKind::LabelKind && "Operand is not a label");
+        return data.label;
+    }
+};
+
+/**
+ * @class InstBase
+ * @brief Represents the base class for instructions in the GIL (Glu Intermediate Language).
+ *
+ * This class provides the basic interface for all instructions, including methods to
+ * get the kind of instruction, its operands, and its results. It also maintains a reference
+ * to the basic block that contains this instruction.
+ *
+ * @note This is an abstract class and cannot be instantiated directly.
+ */
+class InstBase {
```

Comment:
For the use of ilist in BasicBlock, I need your class to inherit from that, can you add it? or you'll merge your PR before mine and I'll do it
```suggestion
class InstBase : public llvm::ilist_node_with_parent<InstBase, BasicBlock> {
```

Reply:
you can do it after merge